Design.md

```diff
@@ -27,7 +27,106 @@ Therefore, v2 will continue to be supported. A user must be able to mix and matc
 
 ### Phobos v3
 
-#### Will use a different package prefix for it, `std3`, thus avoiding any name conflicts with v2
+#### Existing Phobos V2 roots remain reserved indefinitely
+
+The existing roots of `core` for the Runtime, `etc` for C Interfaces, and `std` for the V2 Standard Library are to remain reserved for the foreseeable future. Both V2 and V3 will share usage of the `core` and `etc` roots. The `std` root is to be maintained for compatibility but no new features will be added.
+
+#### Use multiple package prefixes/roots
+
+Phobos V3 will use multiple package roots. This allows us to keep the old `std` package root while splitting up the new library into smaller, more manageable, components. Splitting the library into multiple roots provides the following benefits:
+
+1. Only pay for what you use. Different roots can be compiled as separate static/shared libraries and linked as needed, reducing the overall weight of executables. This offers some relief to the long standing community request to break Phobos into individual packages. While we do not agree with atomizing Phobos to the extreme degree of one package per module, multiple package roots allow us to achieve some of that goal in a logical manner.
+2. Multiple roots allow for layering components. The core roots form a foundation upon which to build higher-level packages.
+3. Multiple roots allow for an expanded feature set without adding to package depth.
+4. Not all roots need to be implemented for a given platform to be considered "supported". If only the core roots are required for a platform to be considered supported it becomes significantly less complicated to port D to new platforms.
+5. Rules can be applied differently to core vs. non-core roots. For example, a possible rule could be that core roots may not throw exceptions under any circumstance, but non-core roots can. This is not intended to imply that throwing exceptions is encouraged, merely made available to allow more implementation flexibility for more complex constructs.
+6. The old `std` root can continue to be maintained and built independently of the new Phobos roots.
+
+Currently the core roots for Phobos V3 are `core`, `etc`, and `sys`. As a rule, Core roots *MUST NOT* import from non-core roots.
+
+Proposed Package Structure (Existing Modules Only):
+```
+core.*
+etc.*
+std.*
+sys
```

Comment:
Everything currently in std moves to sys?

Reply:
At this point, _most_ things move into sys, but there are new roots at the bottom of the list:
data.*
text.*
net.*

I can move them to the top if that makes it easier to read.

One major benefit of this is that all the Unicode stuff gets moved to a new root which will significantly slim down the sys.* root.

Reply:
I think that having too many roots is probably not a good idea since it increases the risk of clashing with user-named packages. Like keywords, in a way.

Reply:
I think you've hit on an important point here. Roots are effectively keywords, and that's why we need to control the common ones. But this is a problem that afflicts any language with some sort of code hierarchy. 

A couple years back MSFT also added the ability to "claim" namespace roots in Nuget for exactly the same reason. And there have been multiple calls on our forums for DUB to be able to do the same. Additionally, it gives a small measure of security in that you know that if you're pulling a package from a specific root, the package that you are getting is _actually_ from that root owner and not somebody trying spoof you into downloading a malicious package. That should be reason enough for us to do the same thing.

But if we're going to do package root pinning then we need to register common root names to the DLF _anyways_ to make sure that people don't try to pass-off their packages as Standard Library packages, and if we have the roots, we might as well use them.